### PR TITLE
4165 - tests with fixtures

### PR DESCRIPTION
**Pull request by Niicck**
https://github.com/cityofaustin/techstack/issues/4165

# Description
Adds tests for our v3 service_page factories.

It also sets a cool pattern for handling test fixtures. I'd love to hear what ya'll think and what improvements we could make on this.

Fixtures for content types can be loaded into:
- in tests
- in your local environment
- as "dummy" data in a deployed environment
  - Using Factory fixtures as LOAD_DATA=dummy data will save us from having to make manual django datadump + migration patches for dummy fixtures.

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?

 

- You can load the dummy data locally with `pipenv run python joplin/manage.py load_test_service_pages`
- Check it out deployed: https://joplin-pr-4165-service-page-te.herokuapp.com/admin/pages/search/

# Checklist:
- [ ] Request reviewers
- [ ] Slack-out message for review
- [ ] Link this PR in the issue
- [ ] Moved card to *review* in zenhub

### CODE REVIEW

File: joplin/pages/service_page/fixtures/helpers/create_fixture.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+def create_fixture(page_data, fixture_name):
+    try:
+        page = ServicePage.objects.get(slug=page_data['slug'])
+    except ServicePage.DoesNotExist:
+        page = None
+    if page:
+        print(f"Skipping {fixture_name}")
+        return None
+
+    page = ServicePageFactory.create(**page_data)
```
Comment (briaguya) [inline joplin/pages/service_page/fixtures/helpers/create_fixture.py]: I like this pattern

File: joplin/pages/service_page/fixtures/test_cases/steps_2.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        "parent": components.home(),
+        "coa_global": False,
+        "title": "Service Page with 2 Steps",
+        "slug": "2-step-yee-haw",
+        "add_topics": {
+            "topics": []
+        },
+        "short_description": "This is a very short description",
+        "dynamic_content": components.dynamic_content,
+        "steps": components.steps_2,
```
Comment (briaguya) [inline joplin/pages/service_page/fixtures/test_cases/steps_2.py]: very cool

File: joplin/pages/service_page/tests.py [hunk 3]
```diff
@@ -2,10 +2,56 @@
...
+    assert isinstance(page, ServicePage)
+    assert page.title == "Service Page with 2 Steps"
+    assert page.slug == "2-step-yee-haw"
+    for i, step in enumerate(page.steps.stream_data):
+        assert step["type"] == expected_steps[i]["type"]
+        assert step["value"] == expected_steps[i]["value"]
+
+
+@pytest.mark.django_db
+def test_create_service_page_with_2_steps():
```
Comment (briaguya) [inline joplin/pages/service_page/tests.py]: this is in here twice? looks like some leftover copypasta maybe?

Comment (Niicck) [reply joplin/pages/service_page/tests.py]: whoops

**Pull request merged**